File: test/ruby/DataTypeInt.rb
#!/usr/bin/env ruby
require 'hyperdex'

def assert &block
    raise RuntimeError unless yield
end

def collapse_iterator(xs)
    s = Set.new []
    while xs.has_next() do
        x = xs.next()
        x.freeze
        s.add(x)
    end
    s
end

def to_objectset(xs)
    s = Set.new []
    xs.each do |x|
        x.freeze
        s.add(x)
    end
    s
end

c = HyperDex::Client::Client.new(ARGV[0], ARGV[1].to_i)
assert { c.put("kv", "k", {}) == true }
assert { c.get("kv", "k") == {:v => 0} }
assert { c.put("kv", "k", {"v" => 1}) == true }
assert { c.get("kv", "k") == {:v => 1} }
assert { c.put("kv", "k", {"v" => -1}) == true }
assert { c.get("kv", "k") == {:v => -1} }
assert { c.put("kv", "k", {"v" => 0}) == true }
assert { c.get("kv", "k") == {:v => 0} }
assert { c.put("kv", "k", {"v" => 9223372036854775807}) == true }
assert { c.get("kv", "k") == {:v => 9223372036854775807} }
assert { c.put("kv", "k", {"v" => -9223372036854775808}) == true }
assert { c.get("kv", "k") == {:v => -9223372036854775808} }

File: test/python/DataTypeInt.py
#!/usr/bin/env python2
import sys
import hyperdex.client
from hyperdex.client import LessEqual, GreaterEqual, LessThan, GreaterThan, Range, Regex, LengthEquals, LengthLessEqual, LengthGreaterEqual
c = hyperdex.client.Client(sys.argv[1], int(sys.argv[2]))
def to_objectset(xs):
    return set([frozenset(x.items()) for x in xs])
assert c.put('kv', 'k', {}) == True
assert c.get('kv', 'k') == {'v': 0}
assert c.put('kv', 'k', {'v': 1}) == True
assert c.get('kv', 'k') == {'v': 1}
assert c.put('kv', 'k', {'v': -1}) == True
assert c.get('kv', 'k') == {'v': -1}
assert c.put('kv', 'k', {'v': 0}) == True
assert c.get('kv', 'k') == {'v': 0}
assert c.put('kv', 'k', {'v': 9223372036854775807}) == True
assert c.get('kv', 'k') == {'v': 9223372036854775807}
assert c.put('kv', 'k', {'v': -9223372036854775808}) == True
assert c.get('kv', 'k') == {'v': -9223372036854775808}
